pandas/core/groupby/groupby.py

```diff
@@ -2047,7 +2051,13 @@ def ngroup(self, ascending: bool = True):
         """
         with _group_selection_context(self):
             index = self._selected_obj.index
-            result = Series(self.grouper.group_info[0], index)
+
+            # GH28330 preserve subclassed Series/DataFrames through calls
```

Comment:
why is this needed?

Reply:
Returning the result of `GroupBy.ngroup()` as a `Series` ends up erasing the subclass information from `result._constructor_expanddim()`, which led to the default return of a `DataFrame`/`Series` instead of the subclass. 

It seemed to me that `self.obj` was either a `Series` or `DataFrame` instance (or subclass of those), depending on if was called by a `SeriesGroupBy` or `DataFrameGroupBy` so the if statement ensures that the proper `Series` (or subclass) is returned.